src/main/java/com/azure/cosmos/kafka/connect/source/CosmosDBSourceTask.java

```diff
@@ -227,11 +227,6 @@ public void stop() {
             changeFeedProcessor.stop();
             changeFeedProcessor = null;
         }
-
-        if (client != null) {
-            client.close();
-            client = null;
-        }
```

Comment:
so we don't have a dispose now to clean up resources that might be open? 
was this what was causing the error? 
now if the connector is stopped and a cosmos client is open, what happens? is it cleaned up properly?

Reply:
This was strangely causing the error. I modeled the approach based on how [Azure-samples did it](https://github.com/Azure-Samples/azure-cosmos-java-sql-api-samples/blob/main/src/main/java/com/azure/cosmos/examples/changefeed/SampleChangeFeedProcessor.java#L105) where they only closed the Changefeedprocessor. Ideally we do want to clean up all resources and I'll do a bit more testing into recreating the connectors when the clients are up.

Reply:
I changed the order of the client close and that didn't make a difference. Tested with deleting and recreating the connector several times and didn't encounter any errors. I suppose we could ask the Cosmos team whether why this behavior is happening.

Reply:
yeah that would be a good idea. let's go ahead and merge this, but not close the issue so we can track this issue with the PG.